src/global_logic.js

```diff
@@ -930,7 +948,145 @@ module.exports.calcHaisuiValue = function (haisuiType, haisuiAmount, haisuiSLv,
+                value = -0.0268 * remainHP + 0.0601;
+            } else {
+                value = -0.1466 * remainHP + 0.12;
+            }
+        } else if (haisuiAmount == "8") {
+            if (remainHP > 0.75 && remainHP <= 1) {
+                value = 0.01;
+            } else if  (remainHP > 0.5 && remainHP <= 0.75) {
+                value = -0.024 * remainHP + 0.068;
+            } else {
+                value = -0.138 * remainHP + 0.125;
+            }
+        } else if (haisuiAmount == "9") {
+            if (remainHP > 0.75 && remainHP <= 1) {
+                value = 0.01;
+            } else if  (remainHP > 0.5 && remainHP <= 0.75) {
+                value = -0.0292 * remainHP + 0.0719;
+            } else {
+                value = -0.1604 * remainHP + 0.1375;
+            }
+        } else if (haisuiAmount == "10") {
+            if (remainHP > 0.75 && remainHP <= 1) {
+                value = 0.01;
+            } else if  (remainHP > 0.5 && remainHP <= 0.75) {
+                value = -0.0332 * remainHP + 0.0749;
+            } else {
+                value = -0.1834 * remainHP + 0.15;
+            } 
+            
+        return value;
+        }
+            
+     if (haisuiType == 'EXLBKonshin' || haisuiType == 'LBKonshin') 
+        if (haisuiAmount == "1") {
+            if (remainHP > 0.6666 && remainHP <= 1) {
+                value = 0.03;
+            } else {
+                value = 0.03 * remainHP + 0.01;
+            }
+        } else if (haisuiAmount == "2") {
+            value = 0.03 * remainHP + 0.01;
+        } else if (haisuiAmount == "3") {
+            if (remainHP > 0.75 && remainHP <= 1) {
+                value = 0.04;
+            } else {
+                value = 0.04 * remainHP + 0.02;
+            }
+        } else if (haisuiAmount == "4") {
+            value = 0.04 * remainHP + 0.02;
+        } else if (haisuiAmount == "5") {
+            if (remainHP > 0.6666 && remainHP <= 1) {
+                value = 0.06;
+            } else {
+                value = 0.06 * remainHP + 0.03;
+            }
+        } else if (haisuiAmount == "6") {
+            if (remainHP > 0.8333 && remainHP <= 1) {
+                value = 0.08;
+            } else {
+                value = 0.06 * remainHP + 0.03;
+            }
+        } else if (haisuiAmount == "7") {
+                value = 0.06 * remainHP + 0.03;
+        } else if (haisuiAmount == "8") {
+            if (remainHP > 0.75 && remainHP <= 1) {
+                value = 0.10;
+            } else {
+                value = 0.08 * remainHP + 0.04;
+            }
+        } else if (haisuiAmount == "9") {
+            if (remainHP > 0.8750 && remainHP <= 1) {
+                value = 0.11;
+            } else {
+                value = 0.08 * remainHP + 0.04;
+            }
+        } else if (haisuiAmount == "10") {
+                value = 0.08 * remainHP + 0.04;
+        }
+    
+        return value;
```

Comment:
This is ok,I just show in comment, haisui 1..9 cases will reach this `return value` (after konshin block).